ush/bash_utils/check_for_preexist_dir_file.sh

```diff
@@ -112,15 +112,42 @@ where the arguments are defined as follows:
 #
 #-----------------------------------------------------------------------
 #
-# If method is set to "overwrite", just overwrite existing directory 
-# file. Useful to keep ongoing run uninterrupted in some situations:
-#   rocotoco *db files and dependent previous cycle dirs stay
+# If method is set to "upgrade", 
+#  keep preexisting directory intact except that
+#  when preexisting directory is $EXPDIR, do the following:
+#    save all old files to a subdirecotry oldxxx/ and then
+#    populate new files into the $EXPDIR directory
+# This is useful to keep ongoing runs uninterrupted:
+#  rocotoco *db files and previous cycles will stay and hence
+#    1. no need to manually cp/mv *db files and previous cycles back
+#    2. no need to manually restart related rocoto tasks failed during
+#         the workflow generation process
+# This may best suit for incremental system upgrades.
+#
+# Alternatively, one can always elect to use the "rename" option 
+#   and then manually do the above aftermath
 #
 #-----------------------------------------------------------------------
 #
-    "overwrite")
+    "upgrade")
+      if [[ "${dir_or_file}" == "${EXPTDIR}" ]]; then
+        local i=1
+        local old_indx=$( printf "%03d" "$i" )
+        local old_dir_or_file="${dir_or_file}/old${old_indx}"
+        while [ -d "${old_dir_or_file}" ]; do
```

Comment:
Does it work when there are already multiple "old" directories, for example, *old003?

Reply:
Yes, if old003/ exists, it will save to old004/; and so on (similar as the "rename" option).